### DISCUSSION TITLE
New home for SDK and DDS plugin

### DISCUSSION THREAD

**Discussion by dennisklein** [FairRootGroup/FairMQ#392] [Announcements]
In this post I would like to recap our journey supporting [DDS🔗](https://dds.gsi.de) as a runtime environment and announce the upcoming deprecation and eventual removal of the DDS plugin and the SDK components in FairMQ.

* In *Q4/2017* - when FairMQ was still a subcomponent in the [FairRoot repository](https://github.com/FairRootGroup/FairRoot) - we released [FairMQ's plugin system](https://github.com/FairRootGroup/FairMQ/blob/bce380d87166530c354b0a376481fcc5e16bdd10/docs/Plugins.md#7-plugins) with [FairRoot `v17.10`](https://github.com/FairRootGroup/FairRoot/releases/tag/v-17.10) which also [shipped the DDS plugin together with its command line interface `fairmq-dds-command-ui`](https://github.com/FairRootGroup/FairMQ/blob/bce380d87166530c354b0a376481fcc5e16bdd10/docs/Plugins.md#731-dds) ever since.
* In *Q3/2019* we released the [FairMQ SDK (namespace `fair::mq::sdk`)](https://fairrootgroup.github.io/FairMQ/v1.4.33/classfair_1_1mq_1_1sdk_1_1BasicTopology.html) with [FairMQ `v1.4.8`](https://github.com/FairRootGroup/FairMQ/releases/tag/v1.4.8). For nostalgia and fun here you can find some early internal presentations on the topic: [Proposal: FairMQ Controller SDK🔗](https://docs.google.com/presentation/d/1BWqLl8sZMq6ewPYUCQGUE4cGjsBf0idFyTJ8CQI5b1Y/edit?usp=sharing), [FairMQ SDK: New Property API🔗](https://docs.google.com/presentation/d/1Bdm004SDSc3N0g_bITbEGEWxBXVZCxn943QrginOfTE/edit?usp=sharing)
* *Up to today* (FairMQ `v1.4.41`) the original implementation of the DDS plugin and its CLI `fairmq-dds-command-ui` have been rewritten and evolved to use the SDK APIs internally. Furthermore, the [ODC project](https://github.com/FairRootGroup/ODC) (newly founded around mid 2019) has been developed and matured significantly. It is the primary consumer of the SDK APIs and the DDS plugin.

The reasons for moving the codebases of the FairMQ SDK and the nowadays tightly coupled DDS plugin to the ODC project are the following:
* **Simplify package dependencies** between FairMQ, DDS, and ODC
   * FairMQ eventually no longer depends on DDS
   * ODC only depends on the rather small and stable FairMQ Plugin APIs instead of the large experimental SDK APIs, which made the release management quite complicated
* **The FairMQ SDK was mainly developed for the ODC use case**, but the original scope of the ODC ("a small stateless adapter") naturally pushed hosting the codebase towards the existing FairMQ repo. The scope and size of the ODC have evolved since then. Today it seems a good fit to keep the FairMQ SDK (which is essentially a client to the DDS plugin using DDS custom commands as a communication channel) codebase development close.
   * The codebase of the **DDS plugin** has to move to the new home naturally, too. To avoid confusion, it was **renamed to `odc`**.
* With the new home in the ODC repo the **design authority** for ongoing and future development now naturally lies within the ODC development team which makes further development a bit more efficient.

## TL;DR 📢

➡️ The FairMQ components `SDK`(experimental), `SDK_COMMANDS`(stable), and `DDS_PLUGIN`(stable) **will be deprecated** in an upcoming `v1.4` patch release and **removed** in an upcoming minor release
➡️ The functionality is superseded by [ODC](https://github.com/FairRootGroup/FairMQ)
➡️ This is merely an organizational change, the FairMQ development team is supporting and developing the relevant codebases in cooperation with the ODC development team just as before!

**What do you need to change?** - From [ODC `v0.30`](https://github.com/FairRootGroup/ODC/releases/tag/0.30) and later we recommend to no longer pass `-P dds` to your FairMQ devices, but `-P odc`.

**Comment by ChristianTackeGSI** [comment]
> The FairMQ components `SDK`(experimental), `SDK_COMMANDS`(stable), and `DDS_PLUGIN`(stable) **will be deprecated** in an upcoming `v1.4` patch release and **removed** in an upcoming minor release

Once the deprecation has happened offcially in FairMQ, I would like to disable exactly those components in FairSoft. Maybe after a small delay (i.e. `sep21` will still ship those components enabled).
